csrc/index_compute.cpp

```diff
@@ -398,6 +460,59 @@ void IndexCompute::handle(Split* split) {
   }
 }
 
+bool IndexCompute::isModuloInvalidUnswitchedIndex(
+    IterDomain* out_concrete_id,
+    Val* out_ind,
+    Val* inner_extent) const {
+  // If not in the unswitched domain map, this domain has no dependent
+  // unswitched domain
+  auto unswitched_domain_map_it = unswitched_domain_map_.find(out_concrete_id);
+  if (unswitched_domain_map_it == unswitched_domain_map_.end()) {
+    return false;
+  }
+
+  for (const auto& unswitched_domain_list : unswitched_domain_map_it->second) {
+    TORCH_INTERNAL_ASSERT(!unswitched_domain_list.empty());
+
+    // If the stride is a multiple of the inner extent, the leaf
+    // unswitched index remains to be a valid maximum index as the
+    // module by the inner extent will be just zero. More
+    // specifically, the index for this unswitched domain would be (x
+    // - 1) * extent_of_inner_domain_0 * extent_of_inner_domain_1
+    // ...., so if the stride component, i.e., the multiplication of all
+    // the inner extents is divisible by the merge inner extent, its
+    // contribution propagated to the inner path will be zero
+    Val* stride = out_concrete_id->fusion()->oneVal();
+    for (auto it = unswitched_domain_list.begin();
+         it != unswitched_domain_list.end() - 1;
+         ++it) {
```

Comment:
Am I understanding correct that the last item of the `deque` is always assigned index `extent - 1`?

Reply:
I think that should be true for most of the cases, but think about:

```
[i0, i1]
-> [i0, i1o, i1i] // split i1
-> [i0, i1i, i1o] // reorder
```

And unswitch only the last domain. The last item for the deque of `i1` would be `i1i`, but since it's not unswitched, the index would not be `extent -1`.

I haven't thought about patterns like this. Wonder if this would break anything...

Reply:
For this case, shouldn't `unswitched_domain_map_` be initialized as
```C++
{ 
  i1o: {{i1o}}
}
```
and `updateUnswitchedDomains` will update it as:
```C++
{
  i1o: {{i1o}},
  i1: {{i1i, i1o}}
}
```
?

Reply:
Oh, sorry, you're right. I think this is correct:

```
the last item of the deque is always assigned index extent - 1
```